src/d3a/d3a_core/simulation.py

```diff
@@ -141,6 +141,7 @@ def _init(self, slowdown, seed, paused, pause_after):
         else:
             random_seed = random.randint(0, 1000000)
             random.seed(random_seed)
+            self.initial_params['seed'] = random_seed
```

Comment:
Could you please assign it on its own variable though, and not use the initial_params for that ? Initial params is also used for defining the initial simulation state, therefore it could have an effect on the random test failures we have encountered. At least this is the only part where nondeterministic behavior could be introduced in the simulation.